### PR TITLE
Improve Anthropic Claude streaming handling

### PR DESCRIPTION
**Pull request by tzolov**
- add StreamEven API domain model for reliably parsing stream events.
 - add StreamHelper#mergeToolUseEvents to aggregate partial tool use jsons into a list of ContentBlocks.
 - add StreamHelper#eventToChatCompletionResponse to convert Flux<StreamEvents> into Flux<ChatCompletionResponse>.
 - Rename MediaContent -> ContentBlock, RequestMessage -> AnthropicMessage, ChatCompletion -> ChatCompletionResponse.
 - Improve tests and docs.

### CODE REVIEW

**Comment by tzolov** [comment]
FYI, @didalgolab

File: models/spring-ai-anthropic/src/main/java/org/springframework/ai/anthropic/api/StreamHelper.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+			}
+
+			if (messageDeltaEvent.usage() != null) {
+				var totalUsage = new Usage(contentBlockReference.get().usage.inputTokens(),
+						messageDeltaEvent.usage().outputTokens());
+				contentBlockReference.get().withUsage(totalUsage);
+			}
+		}
+		else if (event.type().equals(EventType.MESSAGE_STOP)) {
+			System.out.println("MESSAGE_STOP");
```
Comment (didalgolab) [inline models/spring-ai-anthropic/src/main/java/org/springframework/ai/anthropic/api/StreamHelper.java:185]: Remove or use logger?

**Comment by didalgolab** [comment]
@tzolov thanks for the info, very nice work! Any changes requested from my side at this point?

File: models/spring-ai-anthropic/src/main/java/org/springframework/ai/anthropic/api/StreamHelper.java (hunk 1 above)
Comment (tzolov) [reply models/spring-ai-anthropic/src/main/java/org/springframework/ai/anthropic/api/StreamHelper.java:185]: ah, missed this. thanks

**Comment by tzolov** [comment]
Thanks @didalgolab

**Comment by tzolov** [comment]
address review, rebased, squashed and merged at d8be884cb1ed09e1e949a544ec3f9844d4d3a976

**Pull request closed without merging**